### PR TITLE
[SPARTA-748] Add Plugins jar endpoints in order to upload jars and save it into one path published by the API

### PR DESCRIPTION
**Pull request by ahvargas**
After this is possible to add the http uri in the policy definition as userPluginsJars

### CODE REVIEW

File: serving-core/src/main/scala/com/stratio/sparta/serving/core/constants/AppConstant.scala [hunk 1]
```diff
@@ -67,6 +67,9 @@ object AppConstant {
   final val PluginsLocation = "pluginsLocation"
   final val ProvidedPluginsLocation = "provided"
   final val LocalPluginsLocation = "local"
+  final val PluginsPackageLocation = "pluginPackageLocation"
+  final val DefaultPluginsPackageLocation = "/opt/sds/plugins/"
+  final val PluginsURLLocation = "drivers"
```
Comment (compae) [inline serving-core/src/main/scala/com/stratio/sparta/serving/core/constants/AppConstant.scala]: Should be "plugins"

File: serving-api/src/main/scala/com/stratio/sparta/serving/api/actor/ControllerActor.scala [hunk 2]
```diff
@@ -43,66 +43,71 @@ class ControllerActor(actorsMap: Map[String, ActorRef], curatorFramework: Curato
...
-      pathPrefix("driverJar") {
+      pathPrefix(AppConstant.PluginsURLLocation) {
         getFromDirectory(
-          Try(SpartaConfig.getDetailConfig.get.getString(AppConstant.DriverPackageLocation))
-            .getOrElse(AppConstant.DefaultDriverPackageLocation))
-      } ~ pathPrefix("") {
+          Try(SpartaConfig.getDetailConfig.get.getString(AppConstant.PluginsPackageLocation))
+            .getOrElse(AppConstant.DefaultPluginsPackageLocation))
+      } ~
+        pathPrefix("driverJar") {
```
Comment (compae) [inline serving-api/src/main/scala/com/stratio/sparta/serving/api/actor/ControllerActor.scala]: could you add a variable AppConstant.DriverURLLocation??

File: dist/pom.xml [hunk 3]
```diff
@@ -225,6 +225,10 @@
                                 <mkdirs>
                                     <path>/var/log/sds/sparta</path>
                                 </mkdirs>
+                                <!-- For the jars plugin -->
+                                <mkdirs>
+                                    <path>/etc/sds/plugins</path>
```
Comment (compae) [inline dist/pom.xml]: must be /opt/sds/plugins

File: serving-api/src/main/scala/com/stratio/sparta/serving/api/actor/PluginActor.scala [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+class PluginActor extends Actor
+  with Json4sJacksonSupport
+  with SLF4JLogging
+  with SpartaSerializer {
+
+  //The dir where the jars will be saved
+  val targetDir: String = Try(SpartaConfig.getDetailConfig.get.getString(AppConstant.PluginsPackageLocation))
+    .getOrElse(AppConstant.DefaultPluginsPackageLocation)
+  //Configuration of the app
+  val host: String = SpartaConfig.apiConfig.get.getString("host")
```
Comment (compae) [inline serving-api/src/main/scala/com/stratio/sparta/serving/api/actor/PluginActor.scala]: The variables not need type only functions must have the type explicity

**Comment by compae** [comment]
Add configuration variables to docker-entrypoint

File: dist/pom.xml (hunk 3 above)
Comment (ahvargas) [inline dist/pom.xml]: Done

**Pull request merged**